src/docs/arch-fast-truncate.dox

```diff
@@ -0,0 +1,847 @@
+Page deletion information unpacked from the on-disk cell by \c __wt_ref_addr_copy.}
+@row{btree_inline.h, Structure member, \c WT_ADDR_COPY::del_set,
+True if page deletion information was unpacked.}
+@row{btree_inline.h, Hook, in \c __wt_ref_addr_copy,
+Return the page delete information unpacked from the address.}
+@row{btree_inline.h, Function, \c __wt_page_del_visible,
+Function for checking thread visibility of a \c WT_PAGE_DELETED.}
+@row{btree_inline.h, Function, \c __wt_page_del_visible_all,
+Function for checking global visibility of a \c WT_PAGE_DELETED.}
+@row{btree_inline.h, Function, \c __wt_page_del_committed,
+Function for checking whether a \c WT_PAGE_DELETED is committed.}
+@row{btree_inline.h, Function, \c __wt_btcur_skip_page,
+The page-skip function used by cursor next and previous.}
+
+@row{reconcile_inline.h, Hook, in \c __wt_rec_cell_build_addr,
+A hook to choose \c WT_CELL_ADDR_DEL cells when needed and propagate any passed-in page
+deletion information to the packing code.}
+
+@row{timestamp_inline.h, Macro, \c WT_TIME_AGGREGATE_UPDATE_PAGE_DEL,
+Akin to \c WT_TIME_AGGREGATE_UPDATE but for a \c WT_PAGE_DELETED; used in internal page
+reconciliation.}
+
+@row{txn_inline.h, Function, \c __wt_txn_op_delete_apply_prepare_state,
+The code for updating \c WT_PAGE_DELETED at prepare time.  updates to \c WT_PAGE_DELETED.}
+@row{txn_inline.h, Function, \c __wt_txn_op_commit_apply_timestamps,
+Part of the code for updating \c WT_PAGE_DELETED at commit time.
+(The rest is in \c __wt_txn_commit itself.
+Note that the rollback-time update is in bt_delete.c.)}
+@row{txn_inline.h, Hook, in \c __wt_txn_op_set_timestamp,
+Call \c __wt_txn_op_delete_apply_prepare_state and
+\c __wt_txn_op_commit_apply_timestamps.}
+@row{txn_inline.h, Function, \c __wt_txn_modify_page_delete,
+This records a fast-truncate in the current transaction.}
+
+@row{cell_inline.h, Function, \c __cell_page_del_window_cleanup,
+Akin to \c __cell_kv_window_cleanup except for \c WT_PAGE_DELETED.}
+@row{cell_inline.h, Function, \c __cell_redo_page_del_cleanup,
+Function that redoes does the timestamp cleanup for a \c WT_PAGE_DELETED structure; used
+after we bump write generations at the end recovery.}
+@row{cell_inline.h, Hook, in \c __wt_cell_unpack_safe,
+Unpack the page deletion information from deleted-address cells.}
+@row{cell_inline.h, Hook, in \c __wt_cell_pack_addr,
+Pack the page deletion information into deleted-address cells.}
+
+@row{bt_curnext.c, Hook, in \c __wt_btcur_next_prefix,
+Accept a flag argument that sets \c WT_READ_TRUNCATE.}
+@row{bt_curnext.c, Hook, in \c __wt_btcur_next,
+Accept a flag argument that sets \c WT_READ_TRUNCATE.}
+
+@row{bt_curprev.c, Hook, in \c __wt_btcur_prev,
+Accept a flag argument that sets \c WT_READ_TRUNCATE.}
+
+@row{bt_cursor.c:, Function, \c __wt_btcur_range_truncate (and others),
+The cursor-level truncate code lives here.}
+
+@row{bt_debug.c, Hook, in \c __debug_cell_int,
```

Comment:
Should we create a ticket for those two?

Reply:
I guess so. WT-9860